### ISSUE TITLE
Calling SentrySdk.Close() causes OperationCancelledException

### ISSUE THREAD

**Issue by braunsonm** [getsentry/sentry-dotnet#271]
Hi,

When attempting to close Sentry on AppDomain.ProcessExit I get the following exception:
```
System.OperationCanceledException: The operation was canceled.
   at void System.Threading.CancellationToken.ThrowOperationCanceledException()
   at async Task<bool> System.Threading.SemaphoreSlim.WaitUntilCountOrTimeoutAsync(TaskNode asyncWaiter, int millisecondsTimeout, CancellationToken cancellationToken)
   at void System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
```

My code is simply:

```
		AppDomain.CurrentDomain.ProcessExit += (sender, e) =>
		{
			Sentry.SentrySdk.Close();
		};
```

**Comment by bruno-garcia** [comment]
Thanks for raising this @ChaosCA 

Does this exception bubble up from the call to `Close`? Doesn't look like it per your stack trace.

Looks like this is coming from the `SemaphoreSlim` we use at the `BackgroundWorker`:

https://github.com/getsentry/sentry-dotnet/blob/c77736679b500f7cc7ad760d0274f2f703d7b34c/src/Sentry/Internal/BackgroundWorker.cs#L102-L105

Although as you can see in the snippet we do suppress that and initiate the shutdown process of the SDK.

When calling `SentrySdk.Close` we forward the call to the worker which calls `Dispose` on it:
https://github.com/getsentry/sentry-dotnet/blob/c77736679b500f7cc7ad760d0274f2f703d7b34c/src/Sentry/Internal/BackgroundWorker.cs#L246-L275

We do suppress all exceptions and log it. So if the message came from the console, it's not an issue. At that point an `OperationCancelledException` is not strange to show up since it's the mechanism in .NET to cancel asynchronous operations.

Perhaps we could ignore the `LogError` in that case. Like:

Changing this:

https://github.com/getsentry/sentry-dotnet/blob/c77736679b500f7cc7ad760d0274f2f703d7b34c/src/Sentry/Internal/BackgroundWorker.cs#L246

To: 

```csharp
catch (Exception exception) when (exception.GetType() != typeof(OperationCancelledException)
```
Or simply adding a `catch (OperationCancelledException) {}` right before that catch all.

Also, please note the SDK hooks into `ProcessExit` automatically: #242 althought that feature is still on a beta release: `2.0.0-beta3`. It would be great to get it tested btw by the community before we release `2.0.0` (drop the beta from the name) :)

**Comment by ChaosCA** [comment]
I see, we may be seeing this exception in our console because we attach to `AppDomain.CurrentDomain.FirstChanceException` and `AppDomain.CurrentDomain.UnhandledException`.

If I understand what you're saying, this is safe to ignore?

Happy to hear it's triggered automatically in the beta.
Thanks for the comment!

**Comment by bruno-garcia** [comment]
That makes sense. Yes. Either way it's safe to ignore.
We could still modify the code as I mentioned above. It's a great way to get started with the code base here..

**Comment by chan18** [comment]
looks like, by adding 
``` catch (OperationCancelledException) {} ```
will stop the behavior as you said.
``` 
System.OperationCanceledException: The operation was canceled.
   at void System.Threading.CancellationToken.ThrowOperationCanceledException()
   at async Task<bool> System.Threading.SemaphoreSlim.WaitUntilCountOrTimeoutAsync(TaskNode asyncWaiter, int millisecondsTimeout, CancellationToken cancellationToken)
   at void System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
```

**Comment by bruno-garcia** [comment]
This was solved by #408
